subprojects/packagefiles/openssl/generator.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import shutil
+import subprocess
+
+# Change to the directory of the Bash script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+os.chdir(script_dir)
+
+# Node.js version should bundle OpenSSL of matching version to one specified in wrap file
+node_version = "v19.7.0"
+openssl_version = os.environ.get("OPENSSL_VERSION")
+
+if openssl_version is None:
+    with open("../../openssl.wrap", encoding="utf-8") as wrap_file:
+        for line in wrap_file:
+            if "directory = " in line:
+                openssl_version = line.strip().split()[-1]
+                break
+
+# Clone the Node.js repository
+subprocess.run(["git", "clone", "--depth", "1", "-b", node_version, "https://github.com/nodejs/node.git"])
+
+# Remove the generated-config directory
+shutil.rmtree("generated-config", ignore_errors=True)
+
+# Change directory to Node.js deps/openssl
+os.chdir("node/deps/openssl")
+
+# Apply the patch to generate meson.build for different targets
+subprocess.run(["patch", "-u", "config/generate_gypi.pl", "-i", "../../../generate_gypi.pl.patch"])
+
+# Copy meson.build template file
+shutil.copy("../../../meson.build.tmpl", "config/")
+
+# Clone the upstream OpenSSL repository
+shutil.rmtree("openssl")
+subprocess.run(["git", "clone", "--depth", "1", "--branch", "openssl-" + openssl_version, "https://github.com/openssl/openssl.git"])
+
+# Remove the config/archs directory
+shutil.rmtree("config/archs")
+
+# Run the make command with LANG set to C
+os.environ['LANG'] = 'C'
+subprocess.run(["make", "-C", "config"])
+
+# Copy generated files back into the correct place
+find_cmd = 'mkdir -p ../../../generated-$(dirname "$1"); cp "$1" ../../../generated-"$1"'
+subprocess.run(["find", "config/archs", "-name", "meson.build", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-name", "*.asm", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-name", "*.c", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-name", "*.h", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+subprocess.run(["find", "config/archs", "-iname", "*.s", "-exec", "sh", "-c", find_cmd, "_ignored", "{}", ";"])
+
+# AIX is not supported by Meson
+for d in glob.glob("../../../generated-config/archs/aix*"):
+    shutil.rmtree(d)
+
+# 32-bit s390x supported in Meson
+shutil.rmtree("../../../generated-config/archs/linux32-s390x")
+
+# Remove build info files
+for d in glob.glob("../../../generated-config/archs/*/asm_avx2"):
+    shutil.rmtree(d)
+for d in glob.glob("../../../generated-config/archs/*/*/crypto/buildinf.h"):
+    os.remove(d)
```

Comment:
maybe go over a list of globs and remove everything within in a nested loop

Reply:
not sure I follow.